### PR TITLE
Refactor test realm adapter to avoid need for Loader#shimModule

### PR DESCRIPTION
**Pull request by lukemelia**
- All other usages of Loader#shimModule (all in tests) will need to be eliminated before we can remove the method

### CODE REVIEW

**Review by habdelra — COMMENTED** [review]
don't we also need to set the `instance[loadedBy]` in the node adapter too?

**Comment by lukemelia** [comment]
> don't we also need to set the `instance[loadedBy]` in the node adapter too?

@habdelra this PR doesn't shift entirely to using the symbols for looking up a loader. It prefers to use the symbol if the property is set, otherwise falls back to the same method as before.

File: packages/host/tests/helpers/adapter.ts [hunk 1]
```diff
@@ -183,10 +189,6 @@ export class TestRealmAdapter implements RealmAdapter {
       if (typeof value === 'string') {
         fileRefContent = value;
       } else {
-        let moduleURLString = `${this.#paths.url}${path.replace(/\.gts$/, '')}`;
-
-        this.#loader.shimModule(moduleURLString, value as object);
-
```
Comment (habdelra) [inline packages/host/tests/helpers/adapter.ts:189]: so to be clear--the benefit of this PR is that it removes "magic" functionality where we were shimming modules for you behind the scenes, and rather, shimming modules should be an explicit thing that the developer does? (as well as this is a step in the direction of removing shimmed modules altogether?)

Comment (lukemelia) [reply packages/host/tests/helpers/adapter.ts:189]: My hope is that shimming modules is not something we'll need to do on loaders eventually. We can do it for packages using the virtual network handler, and for test realms using the test realm adapter.

**Review by habdelra — APPROVED** [review]

**Comment by lukemelia** [comment]
Closing in favor of #1323

**Pull request closed without merging**